File: arj_intro_py/arj_intro_py/cmd_gen_node.py
import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist
import functools
import time

class CmdGen(Node):

    def __init__(self):
        super().__init__('cmd_gen')
        self.timer = self.create_timer(0.2, self.loop)
        self.cmd_pub = self.create_publisher(Twist, '/turtle1/cmd_vel', 10)
        self.loop_count = 0
        self.get_logger().info("cmd_gen_node [arj_intro_py] has been started")

    def loop(self):
        cmd_msg = Twist()
        if self.loop_count < 20:
            cmd_msg.linear.x = 1.0
            cmd_msg.angular.z = 0.0
        else:
            cmd_msg.linear.x = -1.0
            cmd_msg.angular.z = 1.0
        self.cmd_pub.publish(cmd_msg)
        self.loop_count += 1
        if self.loop_count > 40:
            self.loop_count = 0

def main(args=None):
    rclpy.init(args=args)
    cmd_gen = CmdGen()
    rclpy.spin(cmd_gen)
    cmd_gen.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: arj_intro_cpp/src/cmd_gen_node.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>
#include <math.h>
#include "rclcpp/rclcpp.hpp"
#include "geometry_msgs/msg/twist.hpp"

using namespace std::chrono_literals;
using std::placeholders::_1;

class CmdGen : public rclcpp::Node
{
public:
    CmdGen() : Node("cmd_gen_node")
    {
        timer_ = this->create_wall_timer(std::chrono::milliseconds(200), std::bind(&CmdGen::loop, this));  
        // /turtle1/cmd_vel plublisher
        cmd_pub_ = this->create_publisher<geometry_msgs::msg::Twist>("/turtle1/cmd_vel", 10);
        RCLCPP_INFO(this->get_logger(), "cmd_gen_node [arj_intro_cpp] has been started");

    }

private:

    void loop()
    {
        // Publish transforms
        auto cmd_msg = geometry_msgs::msg::Twist();
        if (loop_count_ < 20)
        {
            cmd_msg.linear.x = 1.0;
            cmd_msg.angular.z = 0.0;
        }
        else
        {
            cmd_msg.linear.x = -1.0;
            cmd_msg.angular.z = 1.0;
        }
        cmd_pub_->publish(cmd_msg);
        loop_count_++;
        if (loop_count_ > 40)
        {
            loop_count_ = 0;
        }
    }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr cmd_pub_;
    int loop_count_ = 0;

};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<CmdGen>());
    rclcpp::shutdown();
    return 0;
}